pkg/services/apiserver/service.go

```diff
@@ -204,8 +206,24 @@ func (s *service) RegisterAPI(b builder.APIGroupBuilder) {
 	s.builders = append(s.builders, b)
 }
 
+func unaryMetadataInjectInterceptor(ctx context.Context) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		spanCtx := trace.SpanContextFromContext(ctx)
+		traceID := spanCtx.TraceID().String()
+		spanID := spanCtx.SpanID().String()
+		// not sure what I need to pass here to get it to propagate
+		ctx = metadata.AppendToOutgoingContext(ctx, "trace_id", traceID)
```

Comment:
I'm just trying stuff here. Not sure what exactly I need to pass in the metadata.